### DISCUSSION TITLE
Is it possible to use latte without composer ?

### DISCUSSION THREAD

**Discussion by danaoairuike** [nette/latte#349] [General]
- Explain your intentions

**Hi, I'm building a little framework ( customized CMS) for myself. And I don't intend to use composer. I have a simple autoloader.
So, I'd like to know that whether I can just copy the directory of https://github.com/nette/latte/tree/master/src/Latte , since it seems Latte does not have any  external dependency.
Thanks**

- It's up to you to make a strong case to convince the project's developers of the merits of this feature

- Please support my efforts in maintaining this project at https://github.com/sponsors/dg

**Comment by MartinMystikJonas** [comment]
It should be possible but you would have to make your own implementation of classmap autoloader or use something like robot loader. Your "simple autoloader" probably would not work. 

Honestly I do not understand why do you want to go for such complications instead of simply using composer.

**Reply by danaoairuike** [reply]
Thank you.

The reason is simple. I'm trying to build a CMS framework, like wordpress.
And I think composer is not for non-developers.

**Reply by MartinMystikJonas** [reply]
You can use composer at build-time yourself and then just ship it to non-developers with vendor dir included. Only one who would have to use composer would be you at build-time and it would simplify your workflow and especially future updates.

**Comment by dg** [comment]
You can use this [autoloader](https://doc.nette.org/en/robot-loader). But Composer is better way to use libraries today.

**Comment by danaoairuike** [comment]
Hello, @MartinMystikJonas 

I just tried to just import the Latte directory, and omit the Tools and Bridge\Tracy directories. The core Latte works fine. Thanks.
I use Psr-4 autoloader.

However, there is a tiny issue.

May I ask what the class " Latte\RuntimeException" is  in the file "https://github.com/nette/latte/blob/master/src/Latte/Loaders/FileLoader.php" ?

I really couldn't find it. I guess it's from nette/utils ? Thanks

**Comment by dg** [comment]
Some classes in Latte are not located according to PSR-4. You'll have to manually require the files `Latte/exceptions.php` a `Latte/attributes.php`

**Reply by danaoairuike** [reply]
Thank you so much
